src/transport.rs

```diff
@@ -72,20 +73,30 @@ impl TransportProtocol {
 
 impl I2cTransport {
     pub fn new(path: &str, address: u16) -> Result<Self> {
+        let last_wake = Instant::now().checked_sub(Duration::from_secs(10)).unwrap();
```

Comment:
Do _not_ use unwraps.. Use an Option<Instant> if you're trying to represent something in the past?